src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_manager.h

```diff
@@ -117,27 +114,30 @@ extern const char* upgrade_error_codes[];
 /**
  * Start listening loop, exits only on error 
  * @param timeout_sec timeout in seconds
+ * @param manager_configs manager configuration parameters
  * @return only on errors, socket will be closed
  * */
-void wm_agent_upgrade_listen_messages(int timeout_sec);
+void wm_agent_upgrade_listen_messages(int timeout_sec, wm_manager_configs manager_configs);
```

Comment:
Since `manager_configs `is used as read-only it could be defined as const. It is also better to use a pointer, to avoid creating a copy of the structure.
`
const wm_manager_configs* manager_configs`